plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/variation-selector/frontend.ts

```diff
@@ -36,13 +36,101 @@ type Context = AddToCartWithOptionsStoreContext & {
 	options: Option[];
 };
 
+// When true, do not handle clicks on attributes, this is to avoid infinite recursion
+let inAutoselectScope = false;
+
 // Set selected pill styles for proper contrast.
 setStyles();
 
 // Stores are locked to prevent 3PD usage until the API is stable.
 const universalLock =
 	'I acknowledge that using a private store means my plugin will inevitably break on the next store release.';
 
+function attributesAutoselect( variation_selectors ) {
+	const { selectedAttributes } =
+		getContext< Context >();
+			// 'pill' is not actually a possible optionStyle value, pills are simply the default option, and they don't give a value to optionStyle
+	for ( const current_variation_selector of variation_selectors ) {
+		const optionStyle = current_variation_selector.dataset?.optionStyle || 'pills';
+		// Dropdown options or Pill inputs,
+		// that HAVE a value (Choose an Option has an empty value of ""),
+		// and are compatible with the possible variations
+		let valid_choices = [];
+		current_variation_selector.querySelectorAll( 'option:not([value=""]), input.wc-block-add-to-cart-with-options-variation-selector-attribute-options__pill-input:not([value=""])' ).forEach( ( el ) => {
+			var name = '';
+			switch ( optionStyle ) {
+				case 'pills':
+					name = el.getAttribute( 'name' );
+					break;
+				case 'dropdown':
+					name = JSON.parse( el.dataset?.wpContext )[ 'name' ];
+					break;
+				default:
+					throw new Error( 'optionStyle not implemented!: ' + optionStyle );
+			}
```

Comment:
Instead of throwing, I would fall back to the `pills` behavior if `optionStyle` doesn't match `dropdown`. It allows simplifying the code and makes the logic a bit less strict:

```suggestion
			const name = 'dropdown' === optionStyle ? JSON.parse( el.dataset?.wpContext )[ 'name' ] : el.getAttribe( 'name' );
```

Reply:
I like it! Done.